src/sentry/seer/code_review/webhook_task.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime, timezone
+from typing import Any
+
+import orjson
+from django.conf import settings
+from urllib3.exceptions import HTTPError
+
+from sentry import options
+from sentry.net.http import connection_from_url
+from sentry.seer.signed_seer_api import make_signed_seer_api_request
+from sentry.silo.base import SiloMode
+from sentry.tasks.base import instrumented_task
+from sentry.taskworker.namespaces import seer_code_review_tasks
+from sentry.taskworker.retry import Retry
+from sentry.utils import metrics
+
+logger = logging.getLogger(__name__)
+
+
+SEER_CODEGEN_PATH = "/v1/automation/codegen"
+# This needs to match the value defined in the Seer API:
+# https://github.com/getsentry/seer/blob/main/src/seer/routes/codegen.py
+SEER_PR_REVIEW_RERUN_PATH = f"{SEER_CODEGEN_PATH}/rerun"
+PREFIX = "seer.code_review.check_run.rerun"
+MAX_RETRIES = 3
+DELAY_BETWEEN_RETRIES = 60  # 1 minute
+
+
+@instrumented_task(
+    name="sentry.seer.code_review.tasks.process_github_webhook_event",
+    namespace=seer_code_review_tasks,
+    retry=Retry(times=MAX_RETRIES, delay=DELAY_BETWEEN_RETRIES, on=(HTTPError,)),
+    silo_mode=SiloMode.REGION,
+    bind=True,  # Bind task instance to access self.request.retries for conditional logging
+)
+def process_github_webhook_event(
+    self: Any, *, organization_id: int, enqueued_at_str: str, **kwargs: Any
+) -> None:
+    """
+    Process GitHub webhook event by forwarding to Seer if applicable.
+
+    This will be expanded to handle other GitHub webhook events in the future.
+
+    Args:
+        self: Task instance (bound via bind=True)
+        organization_id: The organization ID (for logging/metrics)
+        enqueued_at_str: The timestamp when the task was enqueued
+        **kwargs: Additional logging context & other parameters
+    """
+    context = {"organization_id": organization_id, **kwargs}
+    if not options.get("coding_workflows.code_review.github.check_run.rerun.enabled"):
+        logger.info("Skipping rerun request because the option is disabled", extra=context)
+        status = "option_disabled"
+        return
+
+    original_run_id = kwargs.get("original_run_id")
+    if original_run_id is None:
+        metrics.incr(f"{PREFIX}.outcome", tags={"status": "invalid_task_parameters"})
+        raise ValueError("Missing required task parameters.")
+
+    should_record_latency = True
+
+    try:
+        status = make_seer_request(original_run_id)
+    except HTTPError as e:
+        # Catches all urllib3 errors: TimeoutError, MaxRetryError, NewConnectionError,
+        # SSLError, ProxyError, ProtocolError, ResponseError, DecodeError, etc.
+        status = e.__class__.__name__
+        metrics.incr(f"{PREFIX}.outcome", tags={"status": f"error_{status}"})
+        if self.request.retries >= MAX_RETRIES - 1:
+            logger.exception("%s.error", PREFIX, extra=context)
```

Comment:
additionally here -- since we use sentry, we'll have the full stacktrace in the issue -- do we really have to log the status?